src/codeblockui.js

```diff
@@ -0,0 +1,109 @@
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module code-block/codeblockui
+ */
+
+import Plugin from '@ckeditor/ckeditor5-core/src/plugin';
+import Collection from '@ckeditor/ckeditor5-utils/src/collection';
+import Model from '@ckeditor/ckeditor5-ui/src/model';
+import SplitButtonView from '@ckeditor/ckeditor5-ui/src/dropdown/button/splitbuttonview';
+import { createDropdown, addListToDropdown } from '@ckeditor/ckeditor5-ui/src/dropdown/utils';
+import { getLocalizedLanguageDefinitions } from './utils';
+
+import codeBlockIcon from '../theme/icons/codeblock.svg';
+import '../theme/codeblock.css';
+
+/**
+ * The code block UI plugin.
+ *
+ * Introduces the `'codeBlock'` dropdown.
+ *
+ * @extends module:core/plugin~Plugin
+ */
+export default class CodeBlockUI extends Plugin {
+	/**
+	 * @inheritDoc
+	 */
+	init() {
+		const editor = this.editor;
+		const t = editor.t;
+		const componentFactory = editor.ui.componentFactory;
+		const localizedLanguageDefinitions = getLocalizedLanguageDefinitions( editor );
+
+		componentFactory.add( 'codeBlock', locale => {
+			const command = editor.commands.get( 'codeBlock' );
+			const dropdownView = createDropdown( locale, SplitButtonView );
+			const splitButtonView = dropdownView.buttonView;
+
+			splitButtonView.set( {
+				label: t( 'Insert code block' ),
+				tooltip: true,
+				icon: codeBlockIcon,
+				isToggleable: true
+			} );
+
+			splitButtonView.bind( 'isOn' ).to( command, 'value', value => !!value );
+
+			splitButtonView.on( 'execute', () => {
+				editor.execute( 'codeBlock', {
+					language: localizedLanguageDefinitions[ 0 ].class,
+				} );
+
+				editor.editing.view.focus();
+			} );
+
+			dropdownView.on( 'execute', evt => {
+				editor.execute( 'codeBlock', {
+					language: evt.source._codeBlockLanguage,
+					forceValue: true
+				} );
+
+				editor.editing.view.focus();
+			} );
+
+			dropdownView.class = 'ck-code-block-dropdown';
+			dropdownView.bind( 'isEnabled' ).to( command );
+
+			addListToDropdown( dropdownView, this._getLanguageListItemDefinitions( localizedLanguageDefinitions ) );
+
+			return dropdownView;
+		} );
+	}
+
+	/**
+	 * A helper returning a collection of the `codeBlock` dropdown items representing languages
+	 * available for the user to choose from.
+	 *
+	 * @private
+	 * @param {Array.<module:code-block/codeblock~CodeBlockLanguageDefinition>} localizedLanguageDefinitions
+	 * @returns {Iterable.<module:ui/dropdown/utils~ListDropdownItemDefinition>}
+	 */
+	_getLanguageListItemDefinitions( localizedLanguageDefinitions ) {
+		const editor = this.editor;
+		const command = editor.commands.get( 'codeBlock' );
+		const itemDefinitions = new Collection();
+
+		for ( const languageDef of localizedLanguageDefinitions ) {
+			const definition = {
+				type: 'button',
+				model: new Model( {
+					_codeBlockLanguage: languageDef.class,
```

Comment:
I think that we can't name this config value `class` it makes minor WTF here. I don't have better idea but if we'd go with `language-css` that this value woldn't be exaclty what we need here. It might bloat the config but maybe something derivative could work?

```js
const languageConfig = {
    label: 'JavaScript',
    language: 'javascript',
    [class: 'language-javascript']
    // would default to language-${ config.language }` 
    // but can be set to anything...
}
```

Reply:
If by `languageConfig = { ... }` you mean `CodeBlockLanguageDefinition`, then I'm not sure making the interface of the feature more complex for the sake of the internal UI rendering and behavior is a good idea. 

Using the `language` property in the VM (`new Model( { language: languageDef.language, ... }`, I assume you meant this) wouldn't be OK either because in the feature we may have `ListItemView#language` (BiDi?). I'm for using underscored property to make sure this will never collide and it's pretty easy to figure out where it comes from when you log the list item.

Reply:
I posted this comment in wrong location. What I'm worried about is what we put in the configuration object and that we use css class name as a command value and model attribute value. That's why I thought that we can switch the semantics of the configuration to:

```js
ClassicEditor
	.create( editorElement, {
		codeBlock: {
			languages: [
				// or name: 'plaintext'
				{ language: 'plaintext', label: 'Plain text' }, // The default language.
				{ language: 'php', label: 'PHP' },
				{ language: 'java', label: 'Java' },
				{ language: 'javascript', label: 'JavaScript' },
				{ language: 'python', label: 'Python' }
			]
		}
} )
```
So the configuration would configure list of languages and the default CSS class name will be `language-${ config.name }` but you'd always could do:

```js
ClassicEditor
	.create( editorElement, {
		codeBlock: {
			languages: [
				// or name: 'plaintext'
				{ language: 'plaintext', label: 'Plain text', class: 'lang-txt' }, // The default language.
				{ language: 'php', label: 'PHP', class: 'lang-php' },
				{ language: 'java', label: 'Java', class: 'lang-java' },
				{ language: 'javascript', label: 'JavaScript', class: 'lang-javascript' },
				{ language: 'python', label: 'Python', class: 'lang-python' }
			]
		}
} )
```

Reply:
Done in https://github.com/ckeditor/ckeditor5-code-block/pull/1/commits/da2349feee5ba0c0a979737678c81731f721a963.